core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailDataset.java

```diff
@@ -94,4 +118,12 @@ default CloseableIteration<? extends Triple> getTriples(Resource subj, IRI pred,
 		throw new SailException("RDF-star triple retrieval not supported by this store");
 	}
 
+	default Set<StatementOrder> getAvailableOrderings(Resource subj, IRI pred, Value obj, Resource... contexts) {
+		return Set.of();
+	}
+
+	default Comparator<? extends Statement> getComparator(StatementOrder statementOrder) {
+		throw new SailException("Statement ordering not supported by this store");
+	}
```

Comment:
A way to provide a comparator for the Union and Merge Join code. The comparator doesn't need to be the ValueComparator that we have which orders the data based on the SPARQL spec. Most likely the comparator will be something closer to a string based comparator.

Reply:
And I also see now that we need a more generic approach to the comparator because it will need to work on BindingSets too.